CIME/scripts/create_newcase.py

```diff
@@ -231,12 +231,21 @@ def parse_command_line(args, cimeroot, description):
         "--input-dir",
         help="Use a non-default location for input files. This will change the xml value of DIN_LOC_ROOT.",
     )
-    parser.add_argument(
-        "--driver",
-        default=get_cime_default_driver(),
-        choices=("mct", "nuopc", "moab"),
-        help=argparse.SUPPRESS,
-    )
+    if model == "cesm":
+        drv_choices = ("mct", "nuopc")
+    elif model == "e3sm":
+        drv_choices = ("mct", "moab")
+    else:
+        drv_choices = None
+
+    if drv_choices is not None:
```

Comment:
make the help option a variable and set it to help=argparse.SUPPRESS for e3sm and to
"Override the top level driver type and use this one "
            "(changes xml variable COMP_INTERFACE) [this is an advanced option]"
for cesm.

Reply:
Got it.